### PR TITLE
FEATURE: Hide webp files in file list

### PR DESCRIPTION
**Pull request by stephanschuler**
When browsers request jpg or png files and a corresponding webp file
is available the webp file is returned instead. Webp files are created
automatically.

This patch now hides the automatically created webp files in the TYPO3
because having those in place is an implementation detail when it comes
to editors selecting files.

### CODE REVIEW

File: Classes/Fal/Filter/GeneratedFileNamesFilter.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        $itemIdentifier,
+        $parentIdentifier,
+        array $additionalInformation,
+        DriverInterface $driverInstance
+    ) {
+        if (!$driverInstance instanceof LocalDriver) {
+            return true;
+        }
+        if (preg_match('%.+\.(jpe?g|png)\.webp$%i', $itemName)) {
+            return -1;
```
Comment (liwo) [inline Classes/Fal/Filter/GeneratedFileNamesFilter.php:22]: true und -1 als Return? Das ist schon eher schlimm...

Comment (stephanschuler) [reply Classes/Fal/Filter/GeneratedFileNamesFilter.php:22]: Ja, leider.
https://github.com/TYPO3/TYPO3.CMS/blob/TYPO3_8-7/typo3/sysext/core/Classes/Resource/ResourceStorage.php#L724

**Review by liwo — APPROVED** [review]

File: Classes/Fal/Filter/GeneratedFileNamesFilter.php (hunk 1 above)
Comment (liwo) [reply Classes/Fal/Filter/GeneratedFileNamesFilter.php:22]: Ok

**Comment by sievert** [comment]
Scheint auch bei mama zu funzen

**Review by sievert — APPROVED** [review]

**Pull request merged**